// anna-remax-ui/esm/cell/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import * as React from 'react';
import { View, Text } from 'remax/one';
import classNames from 'classnames';
import Icon from '../icon';
import { getPrefixCls } from '../common';
var prefixCls = getPrefixCls('cell');

var Cell = function Cell(props) {
  var label = props.label,
      style = props.style,
      labelStyle = props.labelStyle,
      valueStyle = props.valueStyle,
      children = props.children,
      description = props.description,
      verticality = props.verticality,
      icon = props.icon,
      _props$border = props.border,
      border = _props$border === void 0 ? true : _props$border,
      arrow = props.arrow,
      required = props.required,
      field = props.field,
      _props$defaultNullVal = props.defaultNullValue,
      defaultNullValue = _props$defaultNullVal === void 0 ? '' : _props$defaultNullVal,
      onTap = props.onTap;

  if (verticality) {
    return /*#__PURE__*/React.createElement(View, {
      className: prefixCls,
      style: style,
      onTap: onTap
    }, /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-verticality")
    }, label ? /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-verticality-label"),
      style: labelStyle
    }, required ? /*#__PURE__*/React.createElement(Text, {
      className: "".concat(prefixCls, "-verticality-required")
    }, "*") : null, label) : null, /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-verticality-value"),
      style: valueStyle
    }, children)));
  }

  return /*#__PURE__*/React.createElement(View, {
    className: prefixCls,
    style: style,
    onTap: onTap
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container")
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container-main")
  }, /*#__PURE__*/React.createElement(View, {
    className: classNames("".concat(prefixCls, "-container-main-left"), _defineProperty({}, "".concat(prefixCls, "-container-main-left-input"), field))
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container-main-left-label")
  }, required ? /*#__PURE__*/React.createElement(Text, {
    className: "".concat(prefixCls, "-container-main-left-label-required")
  }, "*") : null, icon ? /*#__PURE__*/React.createElement(Icon, {
    type: icon,
    size: "32px",
    style: {
      marginRight: '10px'
    },
    color: "#333"
  }) : null, label ? /*#__PURE__*/React.createElement(Text, {
    className: "".concat(prefixCls, "-container-main-left-label-value"),
    style: labelStyle
  }, label) : null)), /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container-main-right")
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container-main-right-value"),
    style: valueStyle
  }, children || children === 0 ? children : defaultNullValue), arrow ? /*#__PURE__*/React.createElement(Icon, {
    type: "right",
    style: {
      marginLeft: '10px'
    },
    color: "#666"
  }) : null)), description ? /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container-description")
  }, description) : null), border ? /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-border")
  }) : null);
};

export default Cell;
