// anna-remax-ui/web/cascade/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _classnames = _interopRequireDefault(require("classnames"));

var _icon = _interopRequireDefault(require("../icon"));

var _common = require("../common");

var prefixCls = (0, _common.getPrefixCls)('cascade');
var flag = false;

var Cascade = function Cascade(props) {
  var _classNames;

  var name = props.name,
      _props$value = props.value,
      value = _props$value === void 0 ? [] : _props$value,
      _props$options = props.options,
      options = _props$options === void 0 ? [] : _props$options,
      _props$height = props.height,
      height = _props$height === void 0 ? '80vh' : _props$height,
      _props$className = props.className,
      className = _props$className === void 0 ? '' : _props$className,
      prompt = props.prompt,
      onChange = props.onChange,
      onComplete = props.onComplete;

  var _useState = (0, _react.useState)(options),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      showedOptions = _useState2[0],
      setShowedOptions = _useState2[1];

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      sign = _useState4[0],
      setSign = _useState4[1];

  var _useState5 = (0, _react.useState)(0),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      scrollTop = _useState6[0],
      setScrollTop = _useState6[1]; // useEffect(() => {
  //   if (options && options.length > 0) {
  //     setShowedOptions(options);
  //   }
  // }, [options]) // 无法使用此 useEffect，使用后会报错。


  (0, _react.useEffect)(function () {
    if (sign && value.length > 0 && flag) {
      setSign(false);
      flag = false;
      var item = value[value.length - 1];
      onComplete === null || onComplete === void 0 ? void 0 : onComplete(value, {
        key: item.key,
        value: item.value
      });
    }
  }, [value, sign]);

  var handleClick = /*#__PURE__*/function () {
    var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(i) {
      var nextValue, nextOption;
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              nextValue = [];
              nextOption = {
                key: i.key,
                value: i.value,
                parentKey: i.parentKey
              };

              if (value.length > 0 && value[value.length - 1].parentKey === i.parentKey) {
                nextValue = [].concat((0, _toConsumableArray2["default"])(value.slice(0, value.length - 1)), [nextOption]);
              } else {
                nextValue = [].concat((0, _toConsumableArray2["default"])(value), [nextOption]);
              }

              onChange(nextValue, nextOption);
              setScrollTop(function (top) {
                return top === 0 ? 1 : 0;
              });

              if (Array.isArray(i.children) && i.children.length > 0) {
                setShowedOptions(i.children);
              } else {
                flag = true;
                setSign(true);
              }

            case 6:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));

    return function handleClick(_x) {
      return _ref.apply(this, arguments);
    };
  }();

  var getMatchLevelOptions = function getMatchLevelOptions(key, data) {
    var r = null;

    for (var i = 0; i < data.length; i++) {
      var item = data[i];

      if (key === item.key) {
        r = data;
      } else if (item.children) {
        r = getMatchLevelOptions(key, item.children);
      }

      if (r) {
        return r;
      }
    }

    return r;
  };

  var handleReChoose = function handleReChoose(item, index) {
    onChange(value.slice(0, index + 1));
    setShowedOptions(getMatchLevelOptions(item.key, options));
    setScrollTop(function (top) {
      return top === 0 ? 1 : 0;
    });
  };

  var h = value.length > 1 ? 90 * value.length + 40 - 30 : 130;
  return /*#__PURE__*/_react["default"].createElement(_web.View, {
    className: (0, _classnames["default"])((_classNames = {}, (0, _defineProperty2["default"])(_classNames, prefixCls, true), (0, _defineProperty2["default"])(_classNames, className, true), _classNames)),
    style: {
      height: height
    }
  }, /*#__PURE__*/_react["default"].createElement(_web.View, {
    className: "".concat(prefixCls, "-selected")
  }, value.map(function (item, index) {
    return /*#__PURE__*/_react["default"].createElement(_web.View, {
      key: item.key,
      className: "".concat(prefixCls, "-step"),
      onTap: function onTap() {
        handleReChoose(item, index);
      }
    }, /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: "".concat(prefixCls, "-step-line")
    }, index === value.length - 1 ? null : /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: "".concat(prefixCls, "-step-line-active")
    }), /*#__PURE__*/_react["default"].createElement(_web.Text, {
      className: "".concat(prefixCls, "-step-dot")
    })), /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: "".concat(prefixCls, "-step-container")
    }, /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: "".concat(prefixCls, "-step-content")
    }, /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: "".concat(prefixCls, "-step-left")
    }, item.value), /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: "".concat(prefixCls, "-step-right")
    }, /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: "".concat(prefixCls, "-step-right-text")
    }, prompt === null || prompt === void 0 ? void 0 : prompt(index)), /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: "".concat(prefixCls, "-step-right-arrow")
    }, /*#__PURE__*/_react["default"].createElement(_icon["default"], {
      type: "right",
      size: "24px",
      color: "#999"
    }))))));
  })), /*#__PURE__*/_react["default"].createElement(_web.View, null, /*#__PURE__*/_react["default"].createElement(_web.ScrollView, {
    scrollY: true,
    className: "".concat(prefixCls, "-showed"),
    scrollTop: scrollTop,
    style: {
      height: "calc(".concat(height, " - ").concat(h + 80, "px)")
    }
  }, /*#__PURE__*/_react["default"].createElement(_web.View, {
    className: "".concat(prefixCls, "-showed-title")
  }, "\u9009\u62E9".concat(name)), showedOptions.map(function (i) {
    return /*#__PURE__*/_react["default"].createElement(_web.View, {
      key: i.key,
      className: "".concat(prefixCls, "-showed-category"),
      onTap: function onTap() {
        handleClick(i);
      }
    }, i.value);
  }))));
};

var _default = Cascade;
exports["default"] = _default;
