// anna-remax-ui/esm/picker/index.js
import * as React from 'react';
import { Picker as APicker } from '../one';
import Cell from '../cell';
import FormValue from '../form-value';

var Picker = function Picker(props) {
  var label = props.label,
      border = props.border,
      required = props.required,
      icon = props.icon,
      _props$pickerAlign = props.pickerAlign,
      pickerAlign = _props$pickerAlign === void 0 ? 'left' : _props$pickerAlign,
      options = props.options,
      value = props.value,
      onChange = props.onChange,
      placeholder = props.placeholder,
      children = props.children,
      disabled = props.disabled;

  var handleChangePicker = function handleChangePicker(e) {
    if (e.detail.value < 0) {
      return;
    }

    onChange === null || onChange === void 0 ? void 0 : onChange(options === null || options === void 0 ? void 0 : options[e.detail.value]);
  };

  var valueIndex = (options === null || options === void 0 ? void 0 : options.findIndex(function (item) {
    return item['key'] === value;
  })) || 0;
  var selectedOption = options === null || options === void 0 ? void 0 : options.find(function (option) {
    return option.key === value;
  });
  return /*#__PURE__*/React.createElement(Cell, {
    label: label,
    labelStyle: {
      width: '180px'
    },
    border: border,
    required: required,
    icon: icon,
    field: true
  }, /*#__PURE__*/React.createElement(APicker, {
    "wechat-mode": "selector",
    range: options,
    rangeKey: "value",
    disabled: disabled,
    value: valueIndex,
    onChange: handleChangePicker
  }, children !== null && children !== void 0 ? children : /*#__PURE__*/React.createElement(FormValue, {
    placeholder: placeholder,
    style: {
      textAlign: pickerAlign
    }
  }, selectedOption === null || selectedOption === void 0 ? void 0 : selectedOption.value)));
};

export default Picker;
