// anna-remax-ui/esm/date-picker/index.js
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import _asyncToGenerator from "@babel/runtime/helpers/esm/asyncToGenerator";
import * as React from 'react';
import dayjs from 'dayjs';
import { View } from 'remax/one';
import { datePicker, showToast } from '../one';
import { to, sync } from '../_util';

var DateTime = function DateTime(props) {
  var value = props.value,
      start = props.start,
      _props$format = props.format,
      format = _props$format === void 0 ? 'yyyy-MM-dd' : _props$format,
      placeholder = props.placeholder,
      onChange = props.onChange,
      onOpen = props.onOpen,
      onComplete = props.onComplete;

  var handleClick = /*#__PURE__*/function () {
    var _ref = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
      var _yield$to, _yield$to2, err, res;

      return _regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              onOpen === null || onOpen === void 0 ? void 0 : onOpen();
              _context.next = 3;
              return to(sync(datePicker, {
                format: format,
                currentDate: start ? start : dayjs().format(format),
                complete: onComplete
              }));

            case 3:
              _yield$to = _context.sent;
              _yield$to2 = _slicedToArray(_yield$to, 2);
              err = _yield$to2[0];
              res = _yield$to2[1];

              if (!err) {
                _context.next = 10;
                break;
              }

              showToast({
                type: 'none',
                content: '调用datePicker失败',
                duration: 2000
              });
              return _context.abrupt("return");

            case 10:
              onChange === null || onChange === void 0 ? void 0 : onChange(res.date);

            case 11:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));

    return function handleClick() {
      return _ref.apply(this, arguments);
    };
  }();

  return /*#__PURE__*/React.createElement(View, {
    style: {
      height: '42px'
    },
    onTap: handleClick
  }, value ? value : placeholder);
};

export default DateTime;
