// anna-remax-ui/esm/cascade/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _toConsumableArray from "@babel/runtime/helpers/esm/toConsumableArray";
import _asyncToGenerator from "@babel/runtime/helpers/esm/asyncToGenerator";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import React, { useState, useEffect } from 'react';
import { View, Text } from 'remax/one';
import { ScrollView } from '../one';
import classNames from 'classnames';
import Icon from '../icon';
import { getPrefixCls } from '../common';
var prefixCls = getPrefixCls('cascade');
var flag = false;

var Cascade = function Cascade(props) {
  var _classNames;

  var name = props.name,
      _props$value = props.value,
      value = _props$value === void 0 ? [] : _props$value,
      _props$options = props.options,
      options = _props$options === void 0 ? [] : _props$options,
      _props$height = props.height,
      height = _props$height === void 0 ? '80vh' : _props$height,
      _props$className = props.className,
      className = _props$className === void 0 ? '' : _props$className,
      prompt = props.prompt,
      onChange = props.onChange,
      onComplete = props.onComplete;

  var _useState = useState(options),
      _useState2 = _slicedToArray(_useState, 2),
      showedOptions = _useState2[0],
      setShowedOptions = _useState2[1];

  var _useState3 = useState(false),
      _useState4 = _slicedToArray(_useState3, 2),
      sign = _useState4[0],
      setSign = _useState4[1];

  var _useState5 = useState(0),
      _useState6 = _slicedToArray(_useState5, 2),
      scrollTop = _useState6[0],
      setScrollTop = _useState6[1]; // useEffect(() => {
  //   if (options && options.length > 0) {
  //     setShowedOptions(options);
  //   }
  // }, [options]) // 无法使用此 useEffect，使用后会报错。


  useEffect(function () {
    if (sign && value.length > 0 && flag) {
      setSign(false);
      flag = false;
      var item = value[value.length - 1];
      onComplete === null || onComplete === void 0 ? void 0 : onComplete(value, {
        key: item.key,
        value: item.value
      });
    }
  }, [value, sign]);

  var handleClick = /*#__PURE__*/function () {
    var _ref = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee(i) {
      var nextValue, nextOption;
      return _regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              nextValue = [];
              nextOption = {
                key: i.key,
                value: i.value,
                parentKey: i.parentKey
              };

              if (value.length > 0 && value[value.length - 1].parentKey === i.parentKey) {
                nextValue = [].concat(_toConsumableArray(value.slice(0, value.length - 1)), [nextOption]);
              } else {
                nextValue = [].concat(_toConsumableArray(value), [nextOption]);
              }

              onChange(nextValue, nextOption);
              setScrollTop(function (top) {
                return top === 0 ? 1 : 0;
              });

              if (Array.isArray(i.children) && i.children.length > 0) {
                setShowedOptions(i.children);
              } else {
                flag = true;
                setSign(true);
              }

            case 6:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));

    return function handleClick(_x) {
      return _ref.apply(this, arguments);
    };
  }();

  var getMatchLevelOptions = function getMatchLevelOptions(key, data) {
    var r = null;

    for (var i = 0; i < data.length; i++) {
      var item = data[i];

      if (key === item.key) {
        r = data;
      } else if (item.children) {
        r = getMatchLevelOptions(key, item.children);
      }

      if (r) {
        return r;
      }
    }

    return r;
  };

  var handleReChoose = function handleReChoose(item, index) {
    onChange(value.slice(0, index + 1));
    setShowedOptions(getMatchLevelOptions(item.key, options));
    setScrollTop(function (top) {
      return top === 0 ? 1 : 0;
    });
  };

  var h = value.length > 1 ? 90 * value.length + 40 - 30 : 130;
  return /*#__PURE__*/React.createElement(View, {
    className: classNames((_classNames = {}, _defineProperty(_classNames, prefixCls, true), _defineProperty(_classNames, className, true), _classNames)),
    style: {
      height: height
    }
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-selected")
  }, value.map(function (item, index) {
    return /*#__PURE__*/React.createElement(View, {
      key: item.key,
      className: "".concat(prefixCls, "-step"),
      onTap: function onTap() {
        handleReChoose(item, index);
      }
    }, /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-step-line")
    }, index === value.length - 1 ? null : /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-step-line-active")
    }), /*#__PURE__*/React.createElement(Text, {
      className: "".concat(prefixCls, "-step-dot")
    })), /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-step-container")
    }, /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-step-content")
    }, /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-step-left")
    }, item.value), /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-step-right")
    }, /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-step-right-text")
    }, prompt === null || prompt === void 0 ? void 0 : prompt(index)), /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-step-right-arrow")
    }, /*#__PURE__*/React.createElement(Icon, {
      type: "right",
      size: "24px",
      color: "#999"
    }))))));
  })), /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(ScrollView, {
    scrollY: true,
    className: "".concat(prefixCls, "-showed"),
    scrollTop: scrollTop,
    style: {
      height: "calc(".concat(height, " - ").concat(h + 80, "px)")
    }
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-showed-title")
  }, "\u9009\u62E9".concat(name)), showedOptions.map(function (i) {
    return /*#__PURE__*/React.createElement(View, {
      key: i.key,
      className: "".concat(prefixCls, "-showed-category"),
      onTap: function onTap() {
        handleClick(i);
      }
    }, i.value);
  }))));
};

export default Cascade;
