// anna-remax-ui/esm/input/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import * as React from 'react';
import { Input, View } from 'remax/one';
import classNames from 'classnames';
import Cell from '../cell';
import { getPrefixCls } from '../common';
var prefixCls = getPrefixCls('input');

var AInput = function AInput(props) {
  var _classNames;

  var label = props.label,
      name = props.name,
      type = props.type,
      value = props.value,
      _props$className = props.className,
      className = _props$className === void 0 ? '' : _props$className,
      inputAlign = props.inputAlign,
      placeholder = props.placeholder,
      onChange = props.onChange,
      disabled = props.disabled,
      _props$border = props.border,
      border = _props$border === void 0 ? true : _props$border,
      required = props.required,
      icon = props.icon,
      extra = props.extra,
      maxLength = props.maxLength,
      cellStyle = props.cellStyle;

  var handleChangeInput = function handleChangeInput(e) {
    onChange === null || onChange === void 0 ? void 0 : onChange(e);
  };

  var inputElement = /*#__PURE__*/React.createElement(Input, {
    className: classNames((_classNames = {}, _defineProperty(_classNames, prefixCls, true), _defineProperty(_classNames, "".concat(prefixCls, "-align-right"), inputAlign === 'right'), _defineProperty(_classNames, "".concat(prefixCls, "-align-center"), inputAlign === 'center'), _defineProperty(_classNames, className, true), _classNames)),
    name: name,
    type: type,
    value: value,
    placeholder: placeholder,
    onInput: handleChangeInput,
    disabled: disabled,
    maxLength: maxLength
  });
  return /*#__PURE__*/React.createElement(Cell, {
    style: cellStyle,
    label: label,
    labelStyle: {
      width: '180px'
    },
    border: border,
    required: required,
    icon: icon,
    field: true
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container")
  }, inputElement, extra && /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-extra")
  }, extra)));
};

export default AInput;
