// anna-remax-ui/web/cell/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _classnames = _interopRequireDefault(require("classnames"));

var _icon = _interopRequireDefault(require("../icon"));

var _common = require("../common");

var prefixCls = (0, _common.getPrefixCls)('cell');

var Cell = function Cell(props) {
  var label = props.label,
      style = props.style,
      labelStyle = props.labelStyle,
      valueStyle = props.valueStyle,
      children = props.children,
      description = props.description,
      verticality = props.verticality,
      icon = props.icon,
      _props$border = props.border,
      border = _props$border === void 0 ? true : _props$border,
      arrow = props.arrow,
      required = props.required,
      field = props.field,
      _props$defaultNullVal = props.defaultNullValue,
      defaultNullValue = _props$defaultNullVal === void 0 ? '' : _props$defaultNullVal,
      onTap = props.onTap;

  if (verticality) {
    return /*#__PURE__*/React.createElement(_web.View, {
      className: prefixCls,
      style: style,
      onTap: onTap
    }, /*#__PURE__*/React.createElement(_web.View, {
      className: "".concat(prefixCls, "-verticality")
    }, label ? /*#__PURE__*/React.createElement(_web.View, {
      className: "".concat(prefixCls, "-verticality-label"),
      style: labelStyle
    }, required ? /*#__PURE__*/React.createElement(_web.Text, {
      className: "".concat(prefixCls, "-verticality-required")
    }, "*") : null, label) : null, /*#__PURE__*/React.createElement(_web.View, {
      className: "".concat(prefixCls, "-verticality-value"),
      style: valueStyle
    }, children)));
  }

  return /*#__PURE__*/React.createElement(_web.View, {
    className: prefixCls,
    style: style,
    onTap: onTap
  }, /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-container")
  }, /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-container-main")
  }, /*#__PURE__*/React.createElement(_web.View, {
    className: (0, _classnames["default"])("".concat(prefixCls, "-container-main-left"), (0, _defineProperty2["default"])({}, "".concat(prefixCls, "-container-main-left-input"), field))
  }, /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-container-main-left-label")
  }, required ? /*#__PURE__*/React.createElement(_web.Text, {
    className: "".concat(prefixCls, "-container-main-left-label-required")
  }, "*") : null, icon ? /*#__PURE__*/React.createElement(_icon["default"], {
    type: icon,
    size: "32px",
    style: {
      marginRight: '10px'
    },
    color: "#333"
  }) : null, label ? /*#__PURE__*/React.createElement(_web.Text, {
    className: "".concat(prefixCls, "-container-main-left-label-value"),
    style: labelStyle
  }, label) : null)), /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-container-main-right")
  }, /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-container-main-right-value"),
    style: valueStyle
  }, children || children === 0 ? children : defaultNullValue), arrow ? /*#__PURE__*/React.createElement(_icon["default"], {
    type: "right",
    style: {
      marginLeft: '10px'
    },
    color: "#666"
  }) : null)), description ? /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-container-description")
  }, description) : null), border ? /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-border")
  }) : null);
};

var _default = Cell;
exports["default"] = _default;
