// anna-remax-ui/web/input/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _classnames = _interopRequireDefault(require("classnames"));

var _cell = _interopRequireDefault(require("../cell"));

var _common = require("../common");

var prefixCls = (0, _common.getPrefixCls)('input');

var AInput = function AInput(props) {
  var _classNames;

  var label = props.label,
      name = props.name,
      type = props.type,
      value = props.value,
      _props$className = props.className,
      className = _props$className === void 0 ? '' : _props$className,
      inputAlign = props.inputAlign,
      placeholder = props.placeholder,
      onChange = props.onChange,
      disabled = props.disabled,
      _props$border = props.border,
      border = _props$border === void 0 ? true : _props$border,
      required = props.required,
      icon = props.icon,
      extra = props.extra,
      maxLength = props.maxLength,
      cellStyle = props.cellStyle;

  var handleChangeInput = function handleChangeInput(e) {
    onChange === null || onChange === void 0 ? void 0 : onChange(e);
  };

  var inputElement = /*#__PURE__*/React.createElement(_web.Input, {
    className: (0, _classnames["default"])((_classNames = {}, (0, _defineProperty2["default"])(_classNames, prefixCls, true), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-align-right"), inputAlign === 'right'), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-align-center"), inputAlign === 'center'), (0, _defineProperty2["default"])(_classNames, className, true), _classNames)),
    name: name,
    type: type,
    value: value,
    placeholder: placeholder,
    onInput: handleChangeInput,
    disabled: disabled,
    maxLength: maxLength
  });
  return /*#__PURE__*/React.createElement(_cell["default"], {
    style: cellStyle,
    label: label,
    labelStyle: {
      width: '180px'
    },
    border: border,
    required: required,
    icon: icon,
    field: true
  }, /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-container")
  }, inputElement, extra && /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-extra")
  }, extra)));
};

var _default = AInput;
exports["default"] = _default;
