// anna-remax-ui/web/picker/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _cell = _interopRequireDefault(require("../cell"));

var _formValue = _interopRequireDefault(require("../form-value"));

var Picker = function Picker(props) {
  var label = props.label,
      border = props.border,
      required = props.required,
      icon = props.icon,
      _props$pickerAlign = props.pickerAlign,
      pickerAlign = _props$pickerAlign === void 0 ? 'left' : _props$pickerAlign,
      options = props.options,
      value = props.value,
      onChange = props.onChange,
      placeholder = props.placeholder,
      children = props.children,
      disabled = props.disabled;

  var handleChangePicker = function handleChangePicker(e) {
    if (e.detail.value < 0) {
      return;
    }

    onChange === null || onChange === void 0 ? void 0 : onChange(options === null || options === void 0 ? void 0 : options[e.detail.value]);
  };

  var valueIndex = (options === null || options === void 0 ? void 0 : options.findIndex(function (item) {
    return item['key'] === value;
  })) || 0;
  var selectedOption = options === null || options === void 0 ? void 0 : options.find(function (option) {
    return option.key === value;
  });
  return /*#__PURE__*/React.createElement(_cell["default"], {
    label: label,
    labelStyle: {
      width: '180px'
    },
    border: border,
    required: required,
    icon: icon,
    field: true
  }, /*#__PURE__*/React.createElement(_web.Picker, {
    "wechat-mode": "selector",
    range: options,
    rangeKey: "value",
    disabled: disabled,
    value: valueIndex,
    onChange: handleChangePicker
  }, children !== null && children !== void 0 ? children : /*#__PURE__*/React.createElement(_formValue["default"], {
    placeholder: placeholder,
    style: {
      textAlign: pickerAlign
    }
  }, selectedOption === null || selectedOption === void 0 ? void 0 : selectedOption.value)));
};

var _default = Picker;
exports["default"] = _default;
