// anna-remax-ui/web/date-picker/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var React = _interopRequireWildcard(require("react"));

var _dayjs = _interopRequireDefault(require("dayjs"));

var _web = require("../web");

var _util = require("../_util");

var DateTime = function DateTime(props) {
  var value = props.value,
      start = props.start,
      _props$format = props.format,
      format = _props$format === void 0 ? 'yyyy-MM-dd' : _props$format,
      placeholder = props.placeholder,
      onChange = props.onChange,
      onOpen = props.onOpen,
      onComplete = props.onComplete;

  var handleClick = /*#__PURE__*/function () {
    var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
      var _yield$to, _yield$to2, err, res;

      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              onOpen === null || onOpen === void 0 ? void 0 : onOpen();
              _context.next = 3;
              return (0, _util.to)((0, _util.sync)(_web.datePicker, {
                format: format,
                currentDate: start ? start : (0, _dayjs["default"])().format(format),
                complete: onComplete
              }));

            case 3:
              _yield$to = _context.sent;
              _yield$to2 = (0, _slicedToArray2["default"])(_yield$to, 2);
              err = _yield$to2[0];
              res = _yield$to2[1];

              if (!err) {
                _context.next = 10;
                break;
              }

              (0, _web.showToast)({
                type: 'none',
                content: '调用datePicker失败',
                duration: 2000
              });
              return _context.abrupt("return");

            case 10:
              onChange === null || onChange === void 0 ? void 0 : onChange(res.date);

            case 11:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));

    return function handleClick() {
      return _ref.apply(this, arguments);
    };
  }();

  return /*#__PURE__*/React.createElement(_web.View, {
    style: {
      height: '42px'
    },
    onTap: handleClick
  }, value ? value : placeholder);
};

var _default = DateTime;
exports["default"] = _default;
